policy/src/main/java/org/apache/brooklyn/policy/failover/PropagatePrimaryEnricher.java

```diff
@@ -0,0 +1,208 @@
+    public static final ConfigKey<Collection<? extends Sensor<?>>> PROPAGATING = Propagator.PROPAGATING;
+    public static final ConfigKey<Map<? extends Sensor<?>, ? extends Sensor<?>>> SENSOR_MAPPING = Propagator.SENSOR_MAPPING;
+    
+    //TODO rebind
+    
+    Entity lastPrimary;
+    Propagator propagator;
+    
+    Set<String> effectorsAddedForPrimary;
+    Set<String> blacklistedEffectors = MutableSet.of("start", "stop", "restart", "promote", "demote",
+        getConfig(ElectPrimaryConfig.PROMOTE_EFFECTOR_NAME), getConfig(ElectPrimaryConfig.DEMOTE_EFFECTOR_NAME));
+    Set<String> blacklistedSensors = MutableSet.of();
+    
+    public void setEntity(@SuppressWarnings("deprecation") org.apache.brooklyn.api.entity.EntityLocal entity) {
+        super.setEntity(entity);
+        
+        // TODO if not coming from rebind
+        blacklistedEffectors.addAll(((EntityInternal)entity).getMutableEntityType().getEffectors().keySet());
+        blacklistedSensors.addAll(((EntityInternal)entity).getMutableEntityType().getSensors().keySet());
+        for (Sensor<?> s: Propagator.SENSORS_NOT_USUALLY_PROPAGATED) {
+            blacklistedSensors.add(s.getName());
+        }
+        blacklistedSensors.addAll(MutableSet.of(getConfig(PRIMARY_SENSOR_NAME), getConfig(ElectPrimaryConfig.PRIMARY_WEIGHT_NAME)));
+        
+        subscriptions().subscribe(entity, Sensors.newSensor(Entity.class, config().get(PRIMARY_SENSOR_NAME)), this);
+        onEvent(null);
+    }
+
+    @Override
+    public synchronized void onEvent(SensorEvent<Object> event) {
+        Entity primary = entity.getAttribute( Sensors.newSensor(Entity.class, config().get(PRIMARY_SENSOR_NAME)) );
+        if (!Objects.equal(primary, lastPrimary)) {
+            log.debug("Removing propagated items from "+lastPrimary+" at "+entity);
+            
+            // remove propagator
+            if (propagator!=null) {
+                entity.enrichers().remove(propagator);
+                propagator = null;
+            }
+
+            // remove propagated effectors
+            if (effectorsAddedForPrimary!=null) {
+                log.debug("Removing propagated effectors from "+lastPrimary+" at "+entity+": "+effectorsAddedForPrimary);
+                for (String effN: effectorsAddedForPrimary) {
+                    Effector<?> effE = ((EntityInternal)entity).getMutableEntityType().getEffector(effN);
+                    if (effE!=null) {
+                        ((EntityInternal)entity).getMutableEntityType().removeEffector(effE);
+                    }
+                }
+                effectorsAddedForPrimary = null;
+            }
+            
+            // remove all but blacklisted sensors
+            Set<AttributeSensor<?>> sensorsToRemove = MutableSet.of();
+            for (AttributeSensor<?> s: ((EntityInternal)entity).sensors().getAll().keySet()) {
+                if (!blacklistedSensors.contains(s.getName())) {
```

Comment:
This feels dangerous. When we populate the `blacklistedSensors` in `setEntity` via `entityType.getSensors()`, it will do this very early in the entity's lifecycle. In `setEntity`, the entity may not yet have all the sensors that it will get at runtime.

For example, consider this code:
```
    @Test
    public void testFoo() {
        mgmt.getEntityManager().createEntity(EntitySpec.create(BasicApplication.class)
                .enricher(EnricherSpec.create(MyEnricher.class)));
    }
    public static class MyEnricher extends AbstractEnricher {
        @Override
        public void setEntity(EntityLocal entity) {
            super.setEntity(entity);
            System.out.println(com.google.common.base.Joiner.on(", ").join(((EntityInternal)entity).getMutableEntityType().getSensors().keySet()));
        }
    }
```
The sensors that a `BasicApplication` has when the entity is being instantiated is just:
```
service.isUp, entity.location.removed, entity.config_key.added, service.notUp.indicators, 
entity.children.removed, entity.config_key.removed, entity.group.added, entity.effector.removed, 
entity.sensor.removed, entity.id, entity.location.added, entity.group.removed, catalog.id, 
entity.effector.added, entity.policy.removed, entity.effector.changed, entity.children.added, 
entity.policy.added, entity.sensor.added, application.id
```

It does not yet have sensors like `service.state`, `service.problems`, etc, which are added dynamically.

If I read your code correctly, then these would be automatically removed when the primary changed!

---
I really think we're trying to support too much for an initial version of this policy/enricher. We could simplify things a lot if it was only configurable with `PROPAGATING` and perhaps `SENSOR_MAPPING`.